islandora.module

```diff
@@ -1685,8 +1690,12 @@ function islandora_object_access($op, $object, $user = NULL) {
  */
 function islandora_islandora_object_access($op, $object, $user) {
   module_load_include('inc', 'islandora', 'includes/utilities');
-
-  return islandora_namespace_accessible($object->id) && user_access($op, $user);
+  if ($object->state != 'A') {
+    return islandora_namespace_accessible($object->id) && user_access($op, $user) && user_access(ISLANDORA_ACCESS_INACTIVE_AND_DELETED_OBJECTS, $user);
```

Comment:
> " If an administrator at institution A wants to create a new role they need to be aware of this new permission and grant it."

This is incorrect except in one case. It depends who they consider their "users".
case i. Institution A lets anonymous and authenticated users view islandora objects. (e.g. public archive). Then after this PR, anonymous and authenticated users can view inactive objects. A new role doesn't affect who can view objects, or inactive objects, full stop.
case ii. Institution A only lets authenticated users view islandora objects. Most sites that I know where privacy is an issue use this approach. Again, this pull request will give authenticated users permission to view inactive objects, and life continues as before. New roles don't affect this because all users who have roles are also authenticated users.
case iii. Institution A only lets authenticated users who are also members of a special role view Islandora objects. Maybe they use their islandora site for other, Drupally purposes. Or maybe they let anyone create an account (hi, spambots), so only vetted users get to see the repository. I don't know any sites that do this, but those seem like reasonable if unlikely hypotheticals. This is the only case where this matters. So now we have to imagine that that existing role _viewer-of-islandora-objects_ is insufficient and they need to make another role, who is also exceptionally allowed to view islandora objects. The logical thing to do is to clone the permissions for _viewer-of-islandora-objects_ and assign or remove as needed. I don't know anyone who off the top of their head would know all the complicated permissions that need to be assigned - view, manage objects, ingest objects, add datastreams to objects, etc. So either they're doing it in the GUI by manually cloning the permissions on the original role (which would include this new permission) or by a script.

We do have a status quo of preserving existing functionality, even when it's wrong, like here. What we haven't agreed to, on principle, is to lock ourselves into your vendor scripts from who-knows-how-long-ago. So if you could provide an example, we could see what we're working with, and see how hard it would be (since the claim is that it's an inordinate amount of work) to improve them.